src/Bicep.LangServer/Handlers/BicepDeploymentGraphHandler.cs

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Bicep.Core;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Emit;
+using Bicep.Core.Parsing;
+using Bicep.Core.Resources;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using Bicep.Core.TypeSystem;
+using Bicep.LanguageServer.CompilationManager;
+using Bicep.LanguageServer.Extensions;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Bicep.LanguageServer.Handlers
+{
+    [Method("textDocument/deploymentGraph", Direction.ClientToServer)]
```

Comment:
>deploymentGraph [](http://example.com/codeflow?start=26&length=15)

There isn't anything specific to VS code in this request, right? Any client could implement this if they wanted to?

The reason I'm asking is because of this: [https://microsoft.github.io/language-server-protocol/specifications/specification-current/#dollarRequests](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#dollarRequests)

Reply:
No, there isn't. And yes, any client can implement this method. Technically the client and server can exchange any kinds of custom messages as long as both sides can process them.

Reply:
Sounds good. I was mostly trying to decide if we need the `$` prefix on our message name (`$` prefix means client-specific), but it sounds like we don't.